### PR TITLE
Added SPService resource

### LINKED ISSUE #1235: [New Resource] SPService
### Description

In SharePoint 2016/2019 there is a new feature, service auto provisioning. A few corresponding PowerShell commands also were introduced:

- Get-SPService
- Start-SPService
- Stop-SPService

It would be nice to be able to use DSC to control auto provisioning of one or another service on the farm.

### Proposed properties

Pretty much the same as for
https://github.com/dsccommunity/SharePointDsc/wiki/SPServiceInstance

### Special considerations or limitations

### PR DESCRIPTION
**Pull request by ykuijs**
#### Pull Request (PR) description

Adding new SPService resource to manage provisioning/deprovisioning MinRole services

#### This Pull Request (PR) fixes the following issues

- Fixes #1235 

#### Task list

- [x] Added an entry to the change log under the Unreleased section of the file CHANGELOG.md.
      Entry should say what was changed and how that affects users (if applicable), and
      reference the issue being resolved (if applicable).
- [x] Resource documentation added/updated in README.md.
- [x] Resource parameter descriptions added/updated in README.md, schema.mof and comment-based
      help.
- [x] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309)

### CODE REVIEW

**Review by NikCharlebois — COMMENTED** [review]
One very very minor comment. Also, how would this resource work with SPMinRoleCompliance? If for example, you enforce compliance in SPMinRoleCompliance, but then bypass this with custom SPService declarations, wouldn't that cause the Test function to always return false?

> Reviewed 1 of 7 files at r1, 3 of 10 files at r2, 7 of 7 files at r3.
> *[Reviewable](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309#-:-MXqRVTZ05-VoiUvycDG:b2ju7q3)* status: all files reviewed, 2 unresolved discussions (waiting on @ykuijs)

___
*[SharePointDsc/DSCResources/MSFT_SPService/MSFT_SPService.psm1, line 29 at r3](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309#-MXqQzWa3sHPz4JYxFoo:-MXqQpLeBJpXmwamW4WK:b-sru6gi) ([raw file](https://github.com/dsccommunity/sharepointdsc/blob/3ffa8256ccf822c132c60ae83cb82fd404945580/SharePointDsc/DSCResources/MSFT_SPService/MSFT_SPService.psm1#L29)):*
> ```PowerShell
>     {
>         $message = ("This resource is only supported on SharePoint 2016 and later. " + `
>                 "SharePoint 2013 does not support MinRole yet.")
> ```

Probably picky, but the 'yet' seems to assume it eventually will. 
___
*[SharePointDsc/DSCResources/MSFT_SPService/MSFT_SPService.psm1, line 93 at r3](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309#-MXqR5CBAn3YWeRlveBa:-MXqR5CBAn3YWeRlveBb:bsekjvf) ([raw file](https://github.com/dsccommunity/sharepointdsc/blob/3ffa8256ccf822c132c60ae83cb82fd404945580/SharePointDsc/DSCResources/MSFT_SPService/MSFT_SPService.psm1#L93)):*
> ```PowerShell
>     {
>         $message = ("This resource is only supported on SharePoint 2016 and later. " + `
>                 "SharePoint 2013 does not support MinRole yet.")
> ```

And never will :)

**Review by ykuijs — COMMENTED** [review]
Will update the other two comments.

The Service specifies which services MinRole should check. So if this resource configures the Sandbox code to run, MinRole will make sure that service is running on all WFE servers. If the service instance isn't running on one WFE, the MinRole resource will be incompliant on that server, but this resource is still compliant because the service is still configured to be started in the MinRole config.

So basically this resource is updating the MinRole config which in turn the MinRole resource is checking.

> *[Reviewable](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309#-:-MXqTn7F4q9BaIPmfZa3:b-3uupqv)* status: all files reviewed, 2 unresolved discussions (waiting on @ykuijs)

**Comment by johlju** [comment]
@ykuijs can you change this:

https://github.com/dsccommunity/SharePointDsc/blob/3af354e2d4a0bb2105a292d491977a04e7c93bbe/azure-pipelines.yml#L259-L262

To:

```yml
          - script: |
              bash <(curl -s https://codecov.io/bash) -f "./$(buildFolderName)/$(testResultFolderName)/JaCoCo_coverage.xml"
            displayName: 'Upload to Codecov.io'
```

**Comment by johlju** [comment]
Coverage is not uploading to Codecov.io. I missed that, I was so focused on Azure code coverage. :/

**Comment by ykuijs** [comment]
@johlju Done!

**Review by NikCharlebois — COMMENTED** [review]
One empty line with spaces

> Reviewed 3 of 10 files at r4, 8 of 8 files at r5.
> *[Reviewable](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309#-:-MXs-mNm87yR9UbBgNY9:b92grfx)* status: all files reviewed, 1 unresolved discussion (waiting on @ykuijs)

___
*[SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1, line 285 at r5](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309#-MXs-E371UbZP-IUgJ_L:-MXs-E371UbZP-IUgJ_M:b-k0km2x) ([raw file](https://github.com/dsccommunity/sharepointdsc/blob/7609a4a15fa18fdb77cc3863ad1d5454b7bc6e81/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1#L285)):*
> ```PowerShell
>                 throw $message
>             }
>             
> ```

Remove empty spaces

**Review by ykuijs — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309)* status: 12 of 13 files reviewed, 1 unresolved discussion (waiting on @NikCharlebois)

___
*[SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1, line 285 at r5](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309#-MXs-E371UbZP-IUgJ_L:-MXs2rk38nEbESRBTJFj:b-896fix) ([raw file](https://github.com/dsccommunity/sharepointdsc/blob/7609a4a15fa18fdb77cc3863ad1d5454b7bc6e81/SharePointDsc/DSCResources/MSFT_SPSecurityTokenServiceConfig/MSFT_SPSecurityTokenServiceConfig.psm1#L285)):*
<details><summary><i>Previously, NikCharlebois (Nik Charlebois) wrote…</i></summary><blockquote>

Remove empty spaces
</blockquote></details>

Done.

**Review by NikCharlebois — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 1 files at r6.
> *[Reviewable](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309#-:-MXs2vfe9KpRkp0v8yw4:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ykuijs)

**Review by johlju — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ykuijs)

___
*[azure-pipelines.yml, line 260 at r6](https://reviewable.io/reviews/dsccommunity/sharepointdsc/1309#-MXwEbgMA0O-qXJk1MwK:-MXwEbgMA0O-qXJk1MwL:b7j7t5u) ([raw file](https://github.com/dsccommunity/sharepointdsc/blob/aae3251d83c6f26a7ef7195e4ca812ee43c8859b/azure-pipelines.yml#L260)):*
> ```
> JaCoCo_coverage.xml
> ```

My bad, again. it should be `CodeCov_Merged.xml`. Would be great if you could change it in this PR so we see that it is working. If not, I can fix it in another PR. :slightly_smiling_face:

**Comment by johlju** [comment]
I'm at the summer cabin so I'm trying to do some DSC in-between family time ;)
